internal/util/vault.go

```diff
@@ -89,13 +94,18 @@ func InitVaultKMS(kmsID string, config, secrets map[string]string) (EncryptionKM
 	if !ok || kms.VaultAddress == "" {
 		return nil, fmt.Errorf("missing 'vaultAddress' for vault KMS %s", kmsID)
 	}
-	kms.VaultAuthPath, ok = config["vaultAuthPath"]
-	if !ok || kms.VaultAuthPath == "" {
-		kms.VaultAuthPath = vaultDefaultAuthPath
-	}
-	kms.VaultRole, ok = config["vaultRole"]
-	if !ok || kms.VaultRole == "" {
-		kms.VaultRole = vaultDefaultRole
+
+	kms.VaultToken, ok = config["vaultToken"]
```

Comment:
As both auth methods can be mentioned in the configuration, the preference should be to the stronger method first. IOW, if `vaultAuthPath` and `vaultRole` is defined in addition to `vaultToken` we should not use the token by default.

Reply:
We could do that as well.. Will revert the flow in next patchset.